### PR TITLE
FindSimilarIncidents - fixed bug with numeric fields search 

### PR DESCRIPTION
**Pull request by adi88d**
## XSOAR [Partner](https://xsoar.pan.dev/docs/partners/become-a-tech-partner)?
In case you are a technological partner:
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**.

If you are not a technological partner (i.e. PANW employee, customer or individual developer), please delete this section.

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28540

## Description
Fixed `get_incidents_by_keys` method by changing the query condition to `'%s:=%s'` when the field value is numeric.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py [hunk 1]
```diff
@@ -111,15 +111,14 @@ def get_incident_labels_map(labels):
...
-                                   .replace('\\', '\\\\')
-                                   .replace('"', r'\"')
-                                   .replace("\n", "\\n")
-                                   .replace("\r", "\\r")
-                                   if isinstance(t[1], STRING_TYPES) else t[1]
-                                   ),
-            similar_incident_keys.items()))
+
+    similar_keys_query = []
+    for key in similar_incident_keys.items():
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py:116]: unpack to two variables and use better names
`for x, y in d.items():`

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py [hunk 2]
```diff
@@ -111,15 +111,14 @@ def get_incident_labels_map(labels):
...
+        compare_str = '%s:=%s' if str(key[1]).isdigit() else '%s="%s"'
+        similar_keys_query.append(compare_str % (key[0], str(key[1]).replace('"', r'\"')
+                                                 .replace("\n", "\\n").replace("\r", "\\r")))
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py:119]: don't use `%` string formatting, switch to `.format`.
keep all `replace` calls as in the original code (in separated lines)

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py [hunk 3]
```diff
@@ -111,15 +111,14 @@ def get_incident_labels_map(labels):
...
-        map(lambda t: '%s:"%s"' % (t[0], t[1]
-                                   .replace('\\', '\\\\')
-                                   .replace('"', r'\"')
-                                   .replace("\n", "\\n")
-                                   .replace("\r", "\\r")
-                                   if isinstance(t[1], STRING_TYPES) else t[1]
-                                   ),
-            similar_incident_keys.items()))
+
+    similar_keys_query = []
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py:115]: use different name here, `similar_keys_query` is expected to be a string

**Review by yaakovi — APPROVED** [review]

**Comment by kirbles19** [comment]
@adi88d Done

**Pull request merged**